src/org/zaproxy/zap/model/ValueGenerator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.model;
+
+import java.util.Map;
+
+import org.apache.commons.httpclient.URI;
+
+public interface ValueGenerator {
+	public String getValue(URI uri, String url, String fieldId, Map<String, String> envAttributes, Map<String,String> fieldAttributes);

```

Comment:
- Consider adding JavaDoc (specially to explain the meaning of the parameters).
 - The public modifier could be removed.